src/Core/Houston.vala

```diff
@@ -17,85 +17,154 @@
-            try {
-                var res = process_response ((string) mess.response_body.data);
-                if (res.has_member ("data")) {
-                    var data = res.get_array_member ("data");
+            if (root.has_member ("data")) {
+                var data = root.get_array_member ("data");
+                string[] app_ids = {};
+                data.get_elements ().foreach ((id) => {
+                    app_ids += id.get_string ();
+                });
 
-                    foreach (var id in data.get_elements ()) {
-                        app_ids += ((string) id.get_value ());
-                    }
-                }
-            } catch (Error e) {
-                stderr.printf ("Houston: %s\n", e.message);
+                cache_file.set_string_list (category, APPS_KEY, app_ids);
             }
-            Idle.add (get_newest.callback);
-        });
 
-        yield;
-        return app_ids;
+        } catch (Error e) {
+            throw e;
+        }
     }
 
-    public async string[] get_updated () {
-        var uri = HOUSTON_API_URL + "/newest/release";
-        string[] app_ids = {};
+    private async void refresh () {
+        var uri = HOUSTON_API_URL + "/newest/project";
+        debug ("Requesting newest applications from %s", uri);
+        try {
+            yield process_apps (uri, NEWEST_CATEGORY);
+        } catch (Error e) {
+            critical (e.message);
+        }
 
+        uri = HOUSTON_API_URL + "/newest/release";
         debug ("Requesting recently updated applications from %s", uri);
+        try {
+            yield process_apps (uri, UPDATED_CATEGORY);
+        } catch (Error e) {
+            critical (e.message);
+        }
 
-        var message = new Soup.Message ("GET", uri);
-        session.queue_message (message, (sess, mess) => {
+        lock (cache_file) {
             try {
-                var res = process_response ((string) mess.response_body.data);
-                if (res.has_member ("data")) {
-                    var data = res.get_array_member ("data");
-
-                    foreach (var id in data.get_elements ()) {
-                        app_ids += ((string) id.get_value ());
-                    }
-                }
+                cache_file.save_to_file (get_cache_file_path ());
             } catch (Error e) {
-                stderr.printf ("Houston: %s\n", e.message);
+                critical (e.message);
             }
-            Idle.add (get_updated.callback);
+        }
+
+        refreshed ();
+        uint timeout = (uint)(GLib.TimeSpan.DAY/GLib.TimeSpan.MILLISECOND);
+        GLib.Timeout.add (timeout, () => {
+            refresh.begin ();
+            return GLib.Source.REMOVE;
         });
+    }
 
-        yield;
-        return app_ids;
+    private static string get_cache_file_path () {
+        return GLib.Path.build_filename (GLib.Environment.get_user_cache_dir (), "io.elementary.appcenter", "houston.conf");
```

Comment:
I'm not sure about the ".conf" extension, could this be renamed to something like ".cache" or ".data"?
Regarding the function: could this path constructing go into a `static_construct` to not construct the path everytime we need it?